src/reader/decode/byte_rle.rs

```diff
@@ -0,0 +1,127 @@
+use crate::error::Result;
+use std::io::Read;
+
+use super::util::read_u8;
+
+const MAX_LITERAL_SIZE: usize = 128;
+const MIN_REPEAT_SIZE: usize = 3;
+
+pub struct ByteRleIter<R: Read> {
+    reader: R,
+    literals: [u8; MAX_LITERAL_SIZE],
+    next_byte: Option<u8>,
+    num_literals: usize,
+    used: usize,
+    repeat: bool,
+    min_repeat_size: usize,
+    remaining: usize,
+}
+
+impl<R: Read> ByteRleIter<R> {
+    pub fn new(reader: R, length: usize) -> Self {
+        Self {
+            reader,
+            literals: [0u8; MAX_LITERAL_SIZE],
+            next_byte: None,
+            num_literals: 0,
+            used: 0,
+            repeat: false,
+            min_repeat_size: MIN_REPEAT_SIZE,
```

Comment:
nit: don't need struct field for const, as spec doesn't allow for this min_repeat_size to vary